Make PathautoPatternsForm::buildForm() account for missing config

tstoeckler opened the pull request:
The default pattern config shipped with the module cannot account for alias
types provided by other modules (e.g. File Entity). We cannot use the ternary in
createInstance() directly and instead need to call setConfiguration() because
the defaultConfiguration() would be ignored otherwise.

tstoeckler committed:
Make PathautoPatternsForm::buildForm() account for missing config

The default pattern config shipped with the module cannot account for alias
types provided by other modules (e.g. File Entity). We cannot use the ternary in
createInstance() directly and instead need to call setConfiguration() because
the defaultConfiguration() would be ignored otherwise.

tstoeckler commented:
TBH I think it's less than ideal that setConfiguration() must be called for defaultConfiguration() to be "activated", and I'd love to fix that as well, but not sure whether that should happen here, so I went with the minimally invasive change for now.

Berdir commented:
I believe that's why the BlockBase construct calls setConfiguration(), then we just need to make sure to cast the configuration to an array?

There is unfortunately no base class or something to deal with plugin configuration, so they all do it a bit differently.

tstoeckler committed:
Make EntityAliasTypeBase::__construct() call $this->setConfiguration()

This is inline with BlockBase in core. Without this defaultConfiguration()
is not (always) merged with the passed-in configuration. This allows to
avoid the workaround added in the previous commit.

tstoeckler commented:
Done.

>  then we just need to make sure to cast the configuration to an array

Not sure what you mean, $configuration is type-hinted in the constructor

Berdir commented:
I meant in the caller, the code that you already have. Merged.
